src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs

```diff
@@ -348,6 +648,101 @@ private static double[] ComputeSequenceDeltas(IReadOnlyList<double> a, IReadOnly
             return delta;
         }
 
+        private static ImmutableDictionary<string, TResult>
+            PermutationFeatureImportance<TMetric, TResult>(
+                IHostEnvironment env,
+                ITransformer model,
+                IDataView data,
+                Func<TResult> resultInitializerParam,
+                Func<TMetric, TMetric, TMetric> deltaFuncParam,
+                Func<IDataView, TMetric> evaluationFuncParam,
+                bool useFeatureWeightFilter,
+                int? numberOfExamplesToUse,
+                int permutationCount) where TResult : IMetricsStatistics<TMetric>
+        {
+            env.CheckValue(data, nameof(data));
+            env.CheckValue(model, nameof(model));
+
+            ITransformer lastTransformer = null;
+
+            if (model is TransformerChain<ITransformer> chain)
+            {
+                foreach (var transformer in chain.Reverse())
+                {
+                    if (transformer is ISingleFeaturePredictionTransformer)
+                    {
+                        lastTransformer = transformer;
+                        break;
+                    }
+                }
+            }
+            else lastTransformer = model as ISingleFeaturePredictionTransformer;
+
+            env.CheckValue(lastTransformer, nameof(lastTransformer));
+
+            var lastTransformerType = lastTransformer.GetType();
+            string featureColumnName = (lastTransformer as ISingleFeaturePredictionTransformer).FeatureColumnName;
+            Type pfiType = typeof(PermutationFeatureImportance<,,>);
+
+            TryGetImplementedIPredictionTransformer(lastTransformerType, out lastTransformerType);
+
+            Type[] types = { lastTransformerType.GenericTypeArguments[0], typeof(TMetric), typeof(TResult) };
+            Type pfiGenericType = pfiType.MakeGenericType(types);
+
+            Func<TResult> resultInitializer = resultInitializerParam;
+            Func<IDataView, TMetric> evaluationFunc = idv => evaluationFuncParam.Invoke(idv);
+            Func<TMetric, TMetric, TMetric> deltaFunc = deltaFuncParam;
+
+            object[] param = { env,
+                lastTransformer,
+                data,
+                resultInitializer,
+                evaluationFunc,
+                deltaFunc,
```

Comment:
Why the extra variables here? Why not just pass in the parameters into the array (and then rename the parameters to not have `Param` suffixes).
```suggestion
            object[] param = { env,
                lastTransformer,
                data,
                resultInitializerParam,
                evaluationFuncParam,
                deltaFuncParam,
```

Reply:
I was having an issue with that when I tried it the first time, but I just did it again and its working fine... So it must have been something else. I'll get this updated.